src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -695,11 +702,6 @@ private string GetAssetsFilePath(LockFile lockFile)
 
         private void DowngradeLockFileIfNeeded(LockFile lockFile)
         {
-            if (_request.LockFileVersion <= 2)
-            {
-                DowngradeLockFileToV2(lockFile);
```

Comment:
This one is being addressed in https://github.com/NuGet/NuGet.Client/pull/4020/files#. Not sure which PR is going to be merged.

Reply:
Shouldn't be too big of a deal. 
git should be able to resolve the conflict either way.